File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/mapper/PublishingHouseMapper.java
package ru.elenacherev.librarymanager.mapper;

import ru.elenacherev.librarymanager.api.dto.PublishingHouse;
import ru.elenacherev.librarymanager.domain.entity.PublishingHouseEntity;

public class PublishingHouseMapper {
    public static PublishingHouse map(PublishingHouseEntity entity) {
        return PublishingHouse.builder()
                .publishingHouseId(entity.getPublishingHouseId())
                .title(entity.getTitle())
                .logo(entity.getLogo())
                .build();
    }

    public static PublishingHouseEntity map(PublishingHouse dto) {
        return map(dto, new PublishingHouseEntity());
    }

    public static PublishingHouseEntity map(PublishingHouse dto, PublishingHouseEntity entity) {
        entity.setTitle(dto.getTitle());
        entity.setLogo(dto.getLogo());
        return entity;
    }
}

File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/mapper/PublishingHouseMapper.kt
package ru.elenacherev.librarymanager.mapper

import ru.elenacherev.librarymanager.api.dto.PublishingHouse
import ru.elenacherev.librarymanager.domain.entity.PublishingHouseEntity

fun PublishingHouseEntity.map() = PublishingHouse(
    publishingHouseId = publishingHouseId,
    title = title,
    logo = logo
)

fun PublishingHouse.map() = PublishingHouseEntity(
    title = title,
    logo = logo
)

fun PublishingHouse.map(
    entity: PublishingHouseEntity
) = entity.also {
    it.title = title
    it.logo = logo
}